finch/processes/ensemble_utils.py

```diff
@@ -527,73 +530,87 @@ def ensemble_common_handler(process: Process, request, response, subset_function
+    models = [m.data.strip() for m in request.inputs["models"]]
+    dataset_name = single_input_or_none(request.inputs, "dataset")
 
-    output_filename = make_output_filename(process, request.inputs)
+    base_work_dir = Path(process.workdir)
+    ensembles = []
+    for rcp in rcps:
+        # Ensure no file name conflicts (i.e. if the rcp doesn't appear in the base filename)
+        work_dir = base_work_dir / rcp
+        work_dir.mkdir(exist_ok=True)
+        process.set_workdir(str(work_dir))
+
+        write_log(process, f"Fetching datasets for rcp={rcp}")
+        output_filename = make_output_filename(process, request.inputs, rcp=rcp)
+        netcdf_inputs = get_datasets(
+            dataset_name,
+            workdir=process.workdir,
+            variables=list(source_variable_names),
+            rcp=rcp,
+            models=models,
+        )
 
-    write_log(process, "Fetching datasets")
+        write_log(process, f"Running subset rcp={rcp}", process_step="subset")
 
-    rcp = single_input_or_none(request.inputs, "rcp")
-    models = [m.data.strip() for m in request.inputs["models"]]
-    dataset_name = single_input_or_none(request.inputs, "dataset")
-    netcdf_inputs = get_datasets(
-        dataset_name,
-        workdir=process.workdir,
-        variables=list(source_variable_names),
-        rcp=rcp,
-        models=models,
-    )
+        subsetted_files = subset_function(
+            process, netcdf_inputs=netcdf_inputs, request_inputs=request.inputs
+        )
 
-    write_log(process, "Running subset", process_step="subset")
+        if not subsetted_files:
+            message = "No data was produced when subsetting using the provided bounds."
+            raise ProcessError(message)
 
-    subsetted_files = subset_function(
-        process, netcdf_inputs=netcdf_inputs, request_inputs=request.inputs
-    )
+        subsetted_intermediate_files = compute_intermediate_variables(
+            subsetted_files, dataset_input_names, process.workdir
+        )
 
-    if not subsetted_files:
-        message = "No data was produced when subsetting using the provided bounds."
-        raise ProcessError(message)
+        write_log(process, f"Computing indices rcp={rcp}", process_step="compute_indices")
 
-    subsetted_intermediate_files = compute_intermediate_variables(
-        subsetted_files, dataset_input_names, process.workdir
-    )
+        input_groups = make_indicator_inputs(
+            process.xci, request_inputs_not_datasets, subsetted_intermediate_files
+        )
+        n_groups = len(input_groups)
 
-    write_log(process, "Computing indices", process_step="compute_indices")
+        indices_files = []
 
-    input_groups = make_indicator_inputs(
-        process.xci, request_inputs_not_datasets, subsetted_intermediate_files
-    )
-    n_groups = len(input_groups)
+        warnings.filterwarnings("ignore", category=FutureWarning)
+        warnings.filterwarnings("ignore", category=UserWarning)
```

Comment:
Could you use a context instead ?
E.g. `with warnings.catch_warnings():`

Reply:
haha yes I could, but that's not my code. Simply github not understanding all the diff.

Reply:
After reading on `catch_warnings` it seems it doesn't do much more than what we have. It would be replacing 4 lines by 3+indentation, doesn't feel like a victory. Also it is said to be unsafe in multithreading settings...